marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/impl/CreatedOnFacetHandler.java

```diff
@@ -21,33 +21,113 @@
+            facetInputs.zoneOffset = Integer.parseInt(data.getStringValues().get(1));
+        }
+
+        facetInputs.zoneId = ZoneId.ofOffset("UTC", ZoneOffset.ofTotalSeconds(facetInputs.zoneOffset*60));
+        facetInputs.startDate = LocalDate.now().atStartOfDay(facetInputs.zoneId);
+        facetInputs.endDate = LocalDate.now().atStartOfDay(facetInputs.zoneId);
+
+        switch (facetInputs.timeRange) {
+            case "Today":
+                getTodayTimeWindow(facetInputs);
+                break;
+
+            case "This Week":
+                getThisWeekTimeWindow(facetInputs);
+                break;
+
+            case "This Month":
+                getThisMonthTimeWindow(facetInputs);
+                break;
 
-        // Converting the date in string format from yyyy-MM-dd format to yyyy-MM-dd HH:mm:ss format
-        // Adding 1 day to end date to get docs harmonized on the end date as well.
-        LocalDate endDate = LocalDate.parse(data.getRangeValues().getUpperBound(), DATE_FORMAT)
-                .plusDays(1);
-        String endDateTime = endDate.atStartOfDay(ZoneId.systemDefault()).format(DATE_TIME_FORMAT);
+            case "Custom":
+                getCustomTimeWindow(data, facetInputs);
+                break;
+        }
+        return facetInputs.dateRange;
+    }
+
+    private void getTodayTimeWindow(CreatedOnFacetInputs facetInputs) {
+        facetInputs.startDateTime = facetInputs.startDate.toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("startDateTime", facetInputs.startDateTime);
+        facetInputs.endDateTime = facetInputs.endDate.plusDays(1).toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("endDateTime", facetInputs.endDateTime);
+    }
+
+    private void getThisWeekTimeWindow(CreatedOnFacetInputs facetInputs) {
+        facetInputs.startDateTime = facetInputs.startDate.plusDays((-1) * (facetInputs.startDate.getDayOfWeek().getValue() % 7)).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("startDateTime", facetInputs.startDateTime);
+        facetInputs.endDateTime = facetInputs.endDate.plusDays(1).toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("endDateTime", facetInputs.endDateTime);
+    }
+
+    private void getThisMonthTimeWindow(CreatedOnFacetInputs facetInputs) {
+        facetInputs.startDate = facetInputs.startDate.with(TemporalAdjusters.firstDayOfMonth());
+        facetInputs.startDateTime = facetInputs.startDate.toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("startDateTime", facetInputs.startDateTime);
+        facetInputs.endDateTime = facetInputs.endDate.plusDays(1).toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("endDateTime", facetInputs.endDateTime);
```

Comment:
Method `getTodayTimeWindow` also uses the same code, maybe we can extract common code into a method.

Reply:
No it doesn't. `facetInputs.startDate = facetInputs.startDate.with(TemporalAdjusters.firstDayOfMonth());` 
is used here.
`facetInputs.startDateTime = facetInputs.startDate.toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);`
This is the only common line.

Reply:
I meant lines 99 to 102.